pom.xml

```diff
@@ -140,23 +137,51 @@
                 <version>3.3</version>
                 <configuration>
                     <outputFileName>TotalFreedomMod.jar</outputFileName>
-                    <compilerVersion>1.7</compilerVersion>
-                    <source>1.7</source>
-                    <target>1.7</target>
+                    <compilerVersion>1.8</compilerVersion>
+                    <source>1.8</source>
+                    <target>1.8</target>
                 </configuration>
             </plugin>
 
+            <!-- Git describe -->
+            <plugin>
+                <groupId>com.lukegb.mojo</groupId>
+                <artifactId>gitdescribe-maven-plugin</artifactId>
+                <version>3.0</version>
+                <executions>
+                    <execution>
+                        <id>git-describe</id>
+                        <phase>initialize</phase>
+                        <goals>
+                            <goal>gitdescribe</goal>
+                        </goals>
+                        <configuration>
+                            <descriptionProperty>buildHead</descriptionProperty>
+                            <extraArguments>
+                                <param>--tags</param>
+                                <param>--always</param>
+                                <param>HEAD</param>
+                            </extraArguments>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+            
             <!-- Antrun -->
             <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-antrun-plugin</artifactId>
                 <version>1.8</version>
                 <executions>
                     <execution>
                         <phase>initialize</phase>
                         <configuration>
                             <target>
-                                <propertyfile file="build.properties" comment="Build information. Edit this to your liking.">
+                                <propertyfile file="src/main/resources/build.properties" comment="Build information. Edit this to your liking.">
                                     <entry key="buildAuthor" default="unknown" />
+                                    <entry key="buildNumber" default="0" />
+                                    <entry key="buildCodeName" default="Electrum (MC 1.12) Pre-Release 1" />
```

Comment:
Is there any way not to hardcode the code name in the pom file? Could we instead hard code it in build.properties?

Reply:
These are the build defaults for the POM. You can change them when the build.properties are generated but I wanted to ensure there are always default values as part of the generation.